### PR TITLE
Implement initial option for brace style for non-fn items.

### PR DESCRIPTION
**Pull request by SiegeLord**
This implements this just for enums and structs, as those are the only items I found that are actually have their leading brace formatted (impls, traits and modules are not). I made this a separate option from the brace style for functions for backwards compatibility, but I suppose the function one can be deprecated?

One particular style decision I made was that for empty structs and enums the brace continues to go on the same line as the type, so that formatting like this `struct Foo {}` remains the preferred style. It doesn't seem likely that anyone will prefer this instead:

```rust
struct Foo
{
}
```

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -302,4 +302,5 @@ create_config! {
     take_source_hints: bool, true, "Retain some formatting characteristics from the source code";
     hard_tabs: bool, false, "Use tab characters for indentation, spaces for alignment";
     wrap_comments: bool, false, "Break comments to fit on the line";
+    item_brace_style: BraceStyle, BraceStyle::SameLineWhere, "Brace style for structs and enums";
```
Comment (nrc) [inline src/config.rs]: I'm not sure the ordering makes much difference, but it makes sense to put this next to the function brace style option.

File: tests/source/item-brace-style.rs [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+    struct B {
+        b: i32,
+    }
+
+    // For empty enums and structs, the brace remains on the same line.
+    enum C {}
+
+    struct D {}
+}
```
Comment (nrc) [inline tests/source/item-brace-style.rs]: Could you add some tests with where clauses please?

**Comment by nrc** [comment]
Will this work with `PreferSameLine` and a where clause? It looks like that will still put the brace on a new line rather than the last line of the where clause. In either case, could you add a test for this please?

File: src/config.rs (hunk 1 above)
Comment (SiegeLord) [inline src/config.rs]: Done.

File: tests/source/item-brace-style.rs (hunk 2 above)
Comment (SiegeLord) [inline tests/source/item-brace-style.rs]: Done.

**Comment by SiegeLord** [comment]
Good point, I implemented this somewhat incorrectly. Now I added tests for all 3 different brace styles with and without where clauses.

**Comment by nrc** [comment]
Looks good, thanks for the changes and PR!

**Pull request merged**